src/main/java/net/mcreator/workspace/misc/WorkspaceInfo.java

```diff
@@ -142,17 +141,18 @@ public Map<String, List<MItemBlock>> getCreativeTabMap() {
 		// ModElement#getGeneratableElement that is not thread safe
 		for (GeneratableElement element : elementsList) {
 			if (element instanceof ITabContainedElement tabElement) {
-				TabEntry tabEntry = tabElement.getCreativeTab();
 				List<MCItem> tabItems = tabElement.getCreativeTabItems();
-				if (tabEntry != null && tabItems != null && !tabItems.isEmpty()) {
-					String tab = tabEntry.getUnmappedValue();
-					if (tab != null && !tab.equals("No creative tab entry")) {
-						if (!tabMap.containsKey(tab)) {
-							tabMap.put(tab, new ArrayList<>());
+				if (tabItems != null && !tabItems.isEmpty()) {
+					for (TabEntry tabEntry : tabElement.getCreativeTabs()) {
+						String tab = tabEntry.getUnmappedValue();
+						if (tab != null) {

```

Comment:
Can we expect null values here with your new implementation?

Reply:
There were checks for null value and "absent" tab value, second got removed. I rewrote `getCreativeTab()` but not `getUnmappedValue()` that is checked for `null` here.

Reply:
`tabMap.get(tabEntry.getUnmappedValue()).addAll(<...>)`